### PR TITLE
INF-23 tls and custom middleware support for op-service/rpc

### PR DESCRIPTION
**Pull request by zhwrd**
**Description**

New features for op-service
* adds TLS configuration and https serving support
* adds TLS peer leaf certificate middleware
* adds support for custom http middleware for rpc endpoints

**Tests**

* made sure that PeerTLSMiddleware doesn't break non-https servers
* tested middleware stack manually

**Invariants**

**Additional context**
Used by following implementation:
* https://github.com/ethereum-optimism/infrastructure-services/pull/5

### CODE REVIEW

File: op-service/tls/cli.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+type CLIConfig struct {
+	TLSCaCert string
+	TLSCert   string
+	TLSKey    string
+}
+
+func (c CLIConfig) Check() error {
+
```
Comment (tynes) [inline op-service/tls/cli.go:47]: nit: newline here

File: op-service/tls/tlsinfo.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import (
+	"context"
+	"crypto/x509"
+	"net/http"
+)
+
+type PeerTLSInfo struct {
+	LeafCertificate *x509.Certificate
+}
+
```
Comment (tynes) [inline op-service/tls/tlsinfo.go:12]: Would love some godocs around this PR

Comment (zhwrd) [reply op-service/tls/tlsinfo.go:14]: Added

**Review by tynes — APPROVED** [review]

**Comment by tynes** [comment]
Test coverage is light, the only way to know this works is from running services using the code

**Review by mslipper — APPROVED** [review]

**Pull request merged**